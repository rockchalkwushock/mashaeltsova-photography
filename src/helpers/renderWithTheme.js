import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
