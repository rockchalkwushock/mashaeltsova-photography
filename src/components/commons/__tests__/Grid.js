import React from 'react'

import HomeGrid from '../Grid'
import { renderWithTheme } from '../../../helpers'

describe('Styled Component: <HomeGrid />', () => {
  test('should render markup', () => {
    const wrapper = renderWithTheme(<HomeGrid />)
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    const wrapper = renderWithTheme(<HomeGrid />)
    expect(wrapper).toHaveStyleRule('display', 'grid')
    expect(wrapper).toHaveStyleRule('grid-gap', '20px')
    expect(wrapper).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(200px,300px))'
    )
    expect(wrapper).toHaveStyleRule('justify-content', 'center')
    expect(wrapper).toHaveStyleRule('width', '100%')

    expect(wrapper).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(200px,400px))',
      {
        media: '(min-width:540px)'
      }
    )
    expect(wrapper).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(300px,1fr))',
      {
        media: '(min-width:736px)'
      }
    )
    expect(wrapper).toHaveStyleRule(
      'grid-template-columns',
      'repeat(auto-fit,minmax(200px,1fr))',
      {
        media: '(min-width:900px)'
      }
    )
  })
})
