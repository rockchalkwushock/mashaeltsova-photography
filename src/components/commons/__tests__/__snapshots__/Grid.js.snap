// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <HomeGrid /> should render markup 1`] = `
.c0 {
  grid-gap: 20px;
  grid-template-columns: repeat(auto-fit,minmax(200px,300px));
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

@media (min-width:540px) {
  .c0 {
    grid-template-columns: repeat(auto-fit,minmax(200px,400px));
  }
}

@media (min-width:736px) {
  .c0 {
    grid-template-columns: repeat(auto-fit,minmax(300px,1fr));
  }
}

@media (min-width:900px) {
  .c0 {
    grid-template-columns: repeat(auto-fit,minmax(200px,1fr));
  }
}

<ThemeProvider
  theme={
    Object {
      "accent": "#df744a",
      "bg": "#ffffff",
      "fg": "#bfd8d2",
      "highlight": "#dcb239",
      "lowlight": "#36454f",
      "sizes": Object {
        "lg": "900px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <HomeGrid>
    <div
      className="c0"
    />
  </HomeGrid>
</ThemeProvider>
`;
